src/utils/getUseIsomorphicLayoutEffect.js

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useLayoutEffect } from 'react'
+
+export function getUseIsomorphicLayoutEffect(ssr) {
+  if (typeof ssr === 'boolean') {
+    return ssr ? useEffect : useLayoutEffect;
+  }
+  return (
+    typeof window !== 'undefined' &&
+    typeof window.document !== 'undefined' &&
+    typeof window.document.createElement !== 'undefined'
+      ? useLayoutEffect
+      : useEffect
+  );
```

Comment:
This is great, but can you comment it a little? I think it will be useful for getting this adopted upstream if this explains what is going on and why, like:

`If we have an explicit ssr value, use that to determine if we are server or client-side; otherwise, try to detect SSR state using the existence of the DOM. This inference is not foolproof as some SSR solutions still use a DOM to perform rendering.`